akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -373,302 +372,337 @@ class ActorContextSpec extends TypedSpec(ConfigFactory.parseString(
+          case (subj, child) ⇒
+            val log = muteExpectedException[Exception]("KABOOM2", occurrences = 1)
+            child ! Throw(ex)
+            (subj, child, log)
+        }.expectMessage(expectTimeout) {
+          case (msg, (subj, child, log)) ⇒
+            msg should ===(ChildEvent(GotSignal(PreRestart)))
+            log.assertDone(expectTimeout)
+            child ! BecomeInert(self) // necessary to avoid PostStop/Terminated interference
+            (subj, child)
+        }.expectMessageKeep(expectTimeout) {
+          case (msg, (subj, child)) ⇒
+            msg should ===(BecameInert)
+            stop(subj)
+            ctx.watch(child)
+            ctx.watch(subj)
+        }.expectTermination(expectTimeout) {
+          case (t, (subj, child)) ⇒
+            if (t.ref === child) subj
+            else if (t.ref === subj) child
+            else fail(s"expected termination of either $subj or $child but got $t")
+        }.expectTermination(expectTimeout) { (t, subj) ⇒
+          t.ref should ===(subj)
+        }
+      })
+    }
 
-    def `05 must reset behavior upon Restart`(): Unit = sync(setup("ctx05", Some(Actor.supervise(_).onFailure(SupervisorStrategy.restart))) { (ctx, startWith) ⇒
-      val self = ctx.self
-      val ex = new Exception("KABOOM05")
-      startWith
-        .stimulate(_ ! BecomeInert(self), _ ⇒ BecameInert)
-        .stimulate(_ ! Ping(self), _ ⇒ Pong2) { subj ⇒
-          val log = muteExpectedException[Exception]("KABOOM05")
-          subj ! Throw(ex)
-          subj
+    "stop a child actor" in {
+      sync(setup("ctx04") { (ctx, startWith) ⇒
+        val self = ctx.self
+        startWith.mkChild(Some("A"), ctx.spawnAdapter(ChildEvent), self, inert = true) {
+          case (subj, child) ⇒
+            subj ! Kill(child, self)
+            child
+        }.expectMessageKeep(expectTimeout) { (msg, child) ⇒
+          msg should ===(Killed)
+          ctx.watch(child)
+        }.expectTermination(expectTimeout) { (t, child) ⇒
+          t.ref should ===(child)
         }
-        .stimulate(_ ! Ping(self), _ ⇒ Pong1)
-    })
+      })
+    }
 
-    def `06 must not reset behavior upon Resume`(): Unit = sync(setup(
-      "ctx06",
-      Some(b ⇒ Actor.supervise(b).onFailure(SupervisorStrategy.resume))) { (ctx, startWith) ⇒
+    "reset behavior upon Restart" in {
+      sync(setup("ctx05", Some(Actor.supervise(_).onFailure(SupervisorStrategy.restart))) { (ctx, startWith) ⇒
         val self = ctx.self
-        val ex = new Exception("KABOOM06")
+        val ex = new Exception("KABOOM05")
         startWith
           .stimulate(_ ! BecomeInert(self), _ ⇒ BecameInert)
-          .stimulate(_ ! Ping(self), _ ⇒ Pong2).keep { subj ⇒
-            muteExpectedException[Exception]("KABOOM06", occurrences = 1)
+          .stimulate(_ ! Ping(self), _ ⇒ Pong2) { subj ⇒
+            val log = muteExpectedException[Exception]("KABOOM05")
```

Comment:
Can't see that log is used anywhere?

Reply:
Pfft making me fix things that were already broken! Okay :)